### PR TITLE
Upgrade to okta-auth-js 5.0

### PR DESCRIPTION
**Pull request by denysoblohin-okta**
Issue: [OKTA-386970](https://oktainc.atlassian.net/browse/OKTA-386970)

Version is bumped to 6.0.0 because of changes to authState (initial value is null, isPending is removed).

### CODE REVIEW

**Comment by ryanmr** [comment]
Will this kind of change bump this package's major semver up?

**Comment by shuowu** [comment]
@ryanmr most probably not, as the public Apis of okta-react won't change.

**Comment by denysoblohin-okta** [comment]
I think this should bump major version because of changes to `authState` (initial value is null, `isPending` is removed).

File: README.md [hunk 1]
```diff
@@ -116,7 +117,7 @@ npm install --save @okta/okta-auth-js
...
 - [LoginCallback](#logincallback) - A simple component which handles the login callback when the user is redirected back to the application from the Okta login site.  `<LoginCallback>` accepts an optional prop `errorComponent` that will be used to format the output for any error in handling the callback.  This component will be passed an `error` prop that is an error describing the problem.  (see the `<OktaError>` component for the default rendering)
 
-Users of routers other than `react-router` can use [useOktaAuth](#useoktaauth) to see if a `authState.isPending` is false and `authState.isAuthenticated` is true.  If both are false, you can send them to login via [oktaAuth.signInWithRedirect()](https://github.com/okta/okta-auth-js#signinwithredirectoptions).  See the implementation of `<LoginCallback>` as an example.
+Users of routers other than `react-router` can use [useOktaAuth](#useoktaauth) to see if `authState.isAuthenticated` is true.  If it is false, you can send them to login via [oktaAuth.signInWithRedirect()](https://github.com/okta/okta-auth-js#signinwithredirectoptions).  See the implementation of `<LoginCallback>` as an example.
```
Comment (shuowu) [inline README.md:120]: nit: it would be great to also mention the `null` state here.

**Review by shuowu — APPROVED** [review]

File: src/Security.tsx [hunk 2]
```diff
@@ -61,10 +56,15 @@ const Security: React.FC<{
 
     // Trigger an initial change event to make sure authState is latest
     if (!oktaAuth.isLoginRedirect()) {
-      oktaAuth.authStateManager.updateAuthState();
+      // Calculates initial auth state and fires change event for listeners
```
Comment (aarongranick-okta) [inline src/Security.tsx:59]: Why are we only starting service if not callback? When returning on the callback, the SPA app may go to different routes, when does the service get started in this case? I'm thinking it should start in all cases. What do you think @shuowu ?

File: test/e2e/harness/src/SessionTokenLogin.tsx [hunk 3]
```diff
@@ -29,7 +29,8 @@ const SessionTokenLogin: React.FC = () => {
     })
     .then((res: AuthTransaction) => {
       setSessionToken(res.sessionToken);
-      oktaAuth.signInWithRedirect({sessionToken: res.sessionToken});
+      oktaAuth.setOriginalUri('/');
```
Comment (aarongranick-okta) [inline test/e2e/harness/src/SessionTokenLogin.tsx:32]: not necessary if `restoreOriginalUri` uses `/` as a default  in the case of null

File: test/jest/loginCallback.test.tsx [hunk 4]
```diff
@@ -76,20 +76,11 @@ describe('<LoginCallback />', () => {
       expect(wrapper.text()).toBe('');
     });
 
-    it('does not render errors while authState.isPending', () => {
```
[the remark this answers is not in the document]
Comment (denysoblohin-okta) [reply test/jest/loginCallback.test.tsx:79]: There is a test `does not render errors without an error` which checks rendering when authState is null

File: test/e2e/harness/src/App.tsx [hunk 5]
```diff
@@ -36,7 +36,9 @@ const App: React.FC<{
   };
 
   const restoreOriginalUri = async (_oktaAuth: OktaAuth, originalUri: string) => {
-    history.replace(toRelativeUrl(originalUri, window.location.origin));
+    if (originalUri) {
```
[the remark this answers is not in the document]
Comment (denysoblohin-okta) [reply test/e2e/harness/src/App.tsx]: Fixed

File: src/Security.tsx (hunk 2 above)
Comment (denysoblohin-okta) [reply src/Security.tsx:59]: Added service start after `handleLoginRedirect`

**Review by aarongranick-okta — APPROVED** [review]

**Pull request closed without merging**